--- a/src/itemlistformaction.cpp
+++ b/src/itemlistformaction.cpp
@@ -1252,7 +1252,7 @@
 		auto& item = visible_items[line];
         const bool numbered = (feed->rssurl().find("query:🎬 VODs:") == 0 || feed->rssurl().find("query:📬 New:") == 0 || feed->title() == "⭐ Starred" || feed->title() == "📣 Commentary" || feed->title() == " Favorites") ||
             feed->rssurl().find("query:📥 Downloads:") == 0;
-        const std::string watched = (feed->rssurl().find("query:📥 Downloads:") == 0 || feed->rssurl().find("query:🎬 VODs:") == 0) ? "%4w " : "";
+        const std::string watched = (feed->title() == "⭐ Starred" || feed->rssurl().find("query:📥 Downloads:") == 0 || feed->rssurl().find("query:🎬 VODs:") == 0) ? "%4w " : "";
         const auto format = numbered ? strprintf::fmt("%4u ", line + 1) + watched + itemlist_format : itemlist_format;
         return item2formatted_line(item, width, format, datetime_format);
 	};
